### PR TITLE
Customize HAProxy router config template using a ConfigMap

### PR DESCRIPTION
**Pull request by pecameron**
@knobunc @ramr PTAL

### CODE REVIEW

**Comment by knobunc** [comment]
Can we give this a better name please?

**Comment by knobunc** [comment]
Also, why does this have all of the other changes from the other patch on it?

Can you do a clean branch off master and just put the configmap changes in that branch?  This review is really muddled because it has the other PR in it, and when the other PR merges, this one will then fail.

**Comment by pecameron** [comment]
How about router_configmap? I dont have a bz for this, just trello.
I will redo based on master. There is an additional section with configmap, could you review that?

**Comment by pecameron** [comment]
@knobunc @ramr @adellape  PTAL

File: install_config/install/deploy_router.adoc [hunk 1]
```diff
@@ -875,6 +875,63 @@ endif::[]
 
 Save this content to a file for use as the basis of your customized template.
 
+[[using-configmap]]
+
+=== Using ConfigMap
```
Comment (knobunc) [inline install_config/install/deploy_router.adoc]: "Using a ConfigMap to Replace the Router Configuration Template"

File: install_config/install/deploy_router.adoc [hunk 2]
```diff
@@ -875,6 +875,63 @@ endif::[]
...
+A link:../../dev_guide/configmaps.html[configMap] permits you to customize
+the router instance without rebuilding the router image. The
+*_haproxy-config.template_*, *_reload-haproxy_* and other scripts can be modified
+as well as creating and modifying router environment variables.
+
+The following example shows how to use a custom *_haproxy-config.template_* in
+your router.
+
+First get a copy of the *_haproxy-config.template_* that you want to modify
+as described above. Modify the *_haproxy-config.template_* as desired.
```
Comment (knobunc) [inline install_config/install/deploy_router.adoc]: "described above".  Why not refer to the section?

File: install_config/install/deploy_router.adoc [hunk 3]
```diff
@@ -875,6 +875,63 @@ endif::[]
...
+====
+----
+$ oc create configmap customrouter --from-file=haproxy-config.template
+----
+====
+The configMap contains a copy of your modified *_haproxy-config.template_* file.
+The router dc is next modified to mount the configMap as a file and the `TEMPLATE_FILE`
+environment variable is modified to point to it.
+
+Now edit the router dc to mention the configMap in all the needed places.
```
Comment (knobunc) [inline install_config/install/deploy_router.adoc]: Can we do this with 'oc env' and 'oc volume'?

I think that is easier for people to do.

File: install_config/install/deploy_router.adoc (hunk 1 above)
Comment (pecameron) [inline install_config/install/deploy_router.adoc]: OK

File: install_config/install/deploy_router.adoc (hunk 2 above)
Comment (pecameron) [inline install_config/install/deploy_router.adoc]: Added link.

File: install_config/install/deploy_router.adoc (hunk 3 above)
Comment (pecameron) [inline install_config/install/deploy_router.adoc]: OK

**Comment by pecameron** [comment]
@knobunc I pushed the changes... PTAL

**Comment by knobunc** [comment]
LGTM.  @adellape PTAL

**Comment by pecameron** [comment]
@adellape What needs to be none before you merge this?

**Comment by ramr** [comment]
LGTM.  @adellape / @bfallonf can you PTAL as well and merge.  Thx

**Comment by dcbw** [comment]
I know little about the router, but the instructions make sense to me and I'm pretty sure I could follow them.  LGTM

**Comment by knobunc** [comment]
[merge]

**Comment by tnguyen-rh** [comment]
@pecameron @knobunc @adellape 
I've labeled this as OSE 3.2.  Is that correct?  Additions/corrections welcome.

**Comment by knobunc** [comment]
This works with 3.2, so the label is great if we can get it in those docs.

**Comment by tnguyen-rh** [comment]
@knobunc
Thanks for verifying.
Yeah, in fact the label is the mechanism docs team uses to get stuff from origin to OSE, by cherrypicking from ‘master’ branch to ‘enterprise-3.2’ branch.  (No label, no publishing event.)

**Comment by vikram-redhat** [comment]
@knobunc - Hey Ben, thanks for these docs and the follow up with the labels. To make sure that someone from the docs team looks at these docs before they get published, please don't merge them into the master directly.

**Pull request merged**